sdks/python/apache_beam/ml/inference/base.py

```diff
@@ -587,6 +588,14 @@ def run_inference(
           keys,
           self._unkeyed.run_inference(unkeyed_batch, model, inference_args))
 
+    # The first time a MultiProcessShared ModelManager is used for inference
+    # from this process, we should increment its max model count
+    if self._max_models_per_worker_hint is not None:
+      lock = threading.Lock()
+      if lock.acquire(blocking=False):
```

Comment:
curious why are we not using a blocking lock?

Reply:
The goal is to run this exactly once per process. So the idea is that in a given process, a single thread will acquire the lock and increment the max models. Then the remaining threads (or the original thread in the future) will try to acquire the lock and fail, so they will not increment the value.

Reply:
got it, thanks!